### DISCUSSION TITLE
[Question] TimeoutError on Client Storage

### DISCUSSION THREAD

**Discussion by seriaati** [flet-dev/flet#4009] [Q&A]
### Question

I have an async Flet app and I've noticed that sometimes `TimeoutError` occurs.
For example, on `await page.client_storage.set_async(...)`, error: `Timeout waiting for invokeMethod clientStorage:set`
What does this error mean? Is it safe to be ignored? If not, how should it be handled?
Since I need to access the value being stored later on in the app, the absence of the value might break my app.

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by remohoeppli** [comment]
I see a similar behavior in my app. Especially right after the OAuth login redirects, which happen in the same browser tab using the following command.

`page.login(
    provider,
    on_open_authorization_url=lambda url: page.launch_url(url, web_window_name="_self"),
    redirect_to_page=True
)`

Occasionally, the TimeoutError occurs, other times everything works out smooth. Any help would be much appreciated.

**Reply by seriaati** [reply]
My guess is that either slow internet or slow device hardware caused a long delay between the client storage `setKey` signal and the flet application receiving it. The key is likely already set in the client storage, just that it took the user's end a long time to respond to the application that the key was set, for some reason.
